### PR TITLE
Allow negative amounts in OrderRequest

### LINKED ISSUE #61: CreateOrderRequest does not allow negative discount amount
When I supply negative amounts for the discount line in an order, an exception is thrown indicating this is not allowed. It is though: https://docs.mollie.com/orders/handling-discounts#discounts-and-vouchers.

`lines` parameter supplied to `Gateway::createOrder()`:
```
array:5 [▼
  0 => array:6 [▼
    "name" => "***"
    "unitPrice" => "34.23"
    "quantity" => 3
    "totalAmount" => "102.69"
    "vatRate" => 21
    "vatAmount" => "17.82"
  ]
  1 => array:6 [▼
    "name" => "***"
    "unitPrice" => "9.99"
    "quantity" => 1
    "totalAmount" => "9.99"
    "vatRate" => 6
    "vatAmount" => "0.57"
  ]
  2 => array:6 [▼
    "name" => "***"
    "unitPrice" => "17.50"
    "quantity" => 1
    "totalAmount" => "17.50"
    "vatRate" => 6
    "vatAmount" => "0.99"
  ]
  3 => array:7 [▼
    "name" => "Verzendkosten"
    "unitPrice" => "7.50"
    "quantity" => 1
    "type" => "shipping_fee"
    "totalAmount" => "7.50"
    "vatRate" => 21
    "vatAmount" => "1.30"
  ]
  4 => array:7 [▼
    "name" => "Kortingscode 5EUROKORTING"
    "unitPrice" => "-5.00"
    "quantity" => 1
    "type" => "discount"
    "totalAmount" => "-5.00"
    "vatRate" => 21
    "vatAmount" => "-0.87"
  ]
]
```

Error information:
```
A negative amount is not allowed.

/code/vendor/omnipay/common/src/Common/Message/AbstractRequest.php:342
#0 /code/vendor/omnipay/common/src/Common/Message/AbstractRequest.php(488): Omnipay\Common\Message\AbstractRequest->getMoney('-5.00')
#1 /code/vendor/omnipay/mollie/src/Message/Request/AbstractMollieRequest.php(109): Omnipay\Common\Message\AbstractRequest->formatCurrency('-5.00')
#2 /code/vendor/omnipay/mollie/src/Message/Request/CreateOrderRequest.php(261): Omnipay\Mollie\Message\Request\AbstractMollieRequest->createAmountObject('-5.00')
#3 /code/vendor/omnipay/mollie/src/Message/Request/CreateOrderRequest.php(179): Omnipay\Mollie\Message\Request\CreateOrderRequest->getLines(Object(Omnipay\Common\ItemBag))
#4 /code/vendor/omnipay/common/src/Common/Message/AbstractRequest.php(732): Omnipay\Mollie\Message\Request\CreateOrderRequest->getData()
...: Omnipay\Common\Message\AbstractRequest->send()
```

### PR DESCRIPTION
**Pull request by barryvdh**
Should fix #61

### CODE REVIEW

File: tests/Message/CreateOrderRequestTest.php [hunk 1]
```diff
@@ -159,6 +159,37 @@ public function testGetLines()
...
+                'vatAmount' => '121.14',
+            ],
+            [
+                'type' => 'discount',
+                'name' => 'Discount 100 EURO',
+                'quantity' => 1,
+                'vatRate' => '21.00',
+                'unitPrice' => '-100.00',
+                'totalAmount' => '-100.00',
+                'vatAmount' => '-21.00',
```
Comment (willemstuursma) [inline tests/Message/CreateOrderRequestTest.php]: This is a bit tricky - but for this unit price the `vatAmount` must be `17,36`.

Comment (barryvdh) [reply tests/Message/CreateOrderRequestTest.php]: Okay, shouldn't matter for the test but fixed in https://github.com/thephpleague/omnipay-mollie/commit/ee3604c29cb85ecc9be580558fdcacf8bc994575

Comment (willemstuursma) [reply tests/Message/CreateOrderRequestTest.php]: Right. But we don't want to give anyone the impression that unitPrice or totalAmount are without VAT.

Comment (willemstuursma) [reply tests/Message/CreateOrderRequestTest.php]: Thanks for updating @barryvdh

**Pull request merged**